### ISSUE TITLE
About some definitions in softmax-splatting.py

### ISSUE THREAD

**Issue by Xiaorunyu** [sniklaus/softmax-splatting#33]
Hi Simon,

I have read both of your paper and codes, and that's a great job. But I am confused about part of your codes. I didn't find any definitions of "SIZE_1" ,"VALUE_4" and "OFFSET_4", then how did you work with that?

**Comment by sniklaus** [comment]
Thank you for your kind words. `SIZE_1(tensor)` would be the same as `tensor.shape[1]` in PyTorch, `VALUE_4(tensor, 1, 2, 3, 4)` the same as `tensor[1, 2, 3, 4]`, and `OFFSET_4(tensor, 1, 2, 3, 4)` doesn't have a equivalent in PyTorch as far as I am aware (it gives you the address of a value in the tensor relative to the beginning). These helper functions get implemented through substitution as follows. https://github.com/sniklaus/softmax-splatting/blob/a055eec3c9690095df55d77bbf624ea5fc1d18db/softsplat.py#L179-L232
